spec/linkedlist.js

```diff
@@ -0,0 +1,238 @@
+      expect(myLinkedList.head.data).to.equal("BMW")
+    })
+
+    it('Inserts multiple nodes (with the provided data) to an empty linked list, and checks the value of the head ', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      expect(myLinkedList.tail.data).to.equal("Tesla")
+    })
+
+    it('Inserts multiple nodes (with the provided data) to an empty linked list, and checks the value of the head ', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      expect(myLinkedList.head.next.data).to.equal("Tesla")
+    })
+
+    it('Inserts multiple nodes (with the provided data) to an empty linked list, and checks the value of the head ', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Mercedes")
+      expect(myLinkedList.size).to.equal(3)
+    })
+  })
+
+  context('insertFirst()', () => {
+    it('Inserts nodes (with the provided data) to the front of the linked list and checks the size of the list', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insertFirst("Mercedes")
+      expect(myLinkedList.size).to.equal(3)
+      expect(myLinkedList.head.data).to.equal("Mercedes")
+      expect(myLinkedList.tail.data).to.equal("Tesla")
+    })
+  })
+
+  context('contains(input)', () => {
+    it('Determines whether or not the list contains the provided data', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insertFirst("Mercedes")
+      expect(myLinkedList.contains("Lexus")).to.equal(true)
+      expect(myLinkedList.contains("BMW")).to.equal(true)
+      expect(myLinkedList.contains("Ford")).to.equal(false)
+    })
+  })
+
+  context('find( input )', () => {
+    it('Returns the first node containing the provided data, or -1 if not found', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insert("Ford")
+      myLinkedList.insertFirst("Mercedes")
+      expect(myLinkedList.find("Ford").data).to.equal("Ford")
+      expect(myLinkedList.find("Porsche")).to.equal(-1)
+    })
+  })
+
+  context('insertBefore( before, after )', () => {
+    it('Inserts a node (with data "before") before the first node containing "after"', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insert("Mercedes")
+      myLinkedList.insertBefore("Ford", "BMW")
+      expect(myLinkedList.head.next.data).to.equal("BMW")
```

Comment:
I would include maybe 1 or 2 more expects, checking the size or giving it something that will break it and test for that.